File: Problem_4/problem4.py
def isPalindrome(number):
    return str(number) == str(number)[::-1]

result = 0
for i in range(101, 1000):
    for j in range(101, 1000):
        prod = i*j
        if prod > result and isPalindrome(prod):
            result = prod

print(result)

File: Problem_4/problem4.rs
// rustc problem4.rs

fn is_pal(number: i32) -> bool {
    let string = number.to_string();
    string == string.chars().rev().collect::<String>()
}

fn main() {
    let mut result = 0;

    for i in 101..1000 {
        for j in 101..1000 {
            let prod = i*j;
            if prod > result && is_pal(prod) {
                result = prod;
            }
        }
    }
    println!("{}", result);
}